pkg/kubelet/status/status_manager.go

```diff
@@ -558,8 +558,10 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 
 	// We don't handle graceful deletion of mirror pods.
 	if m.canBeDeleted(pod, status.status) {
-		deleteOptions := metav1.NewDeleteOptions(0)
-		// Use the pod UID as the precondition for deletion to prevent deleting a newly created pod with the same name and namespace.
+		deleteOptions := metav1.DeleteOptions{}
+		deleteOptions.GracePeriodSeconds = new(int64)
+		// Use the pod UID as the precondition for deletion to prevent deleting a
+		// newly created pod with the same name and namespace.
 		deleteOptions.Preconditions = metav1.NewUIDPreconditions(string(pod.UID))
 		err = m.kubeClient.CoreV1().Pods(pod.Namespace).Delete(context.TODO(), pod.Name, deleteOptions)
```

Comment:
nit: would it be easier to just deref `deleteOptions` here?

Reply:
alternately make the NewDeleteOptions helper return a non-pointer since that is what is needed everywhere now